### PR TITLE
ParentBasedSampler should not explicitly consider Links

### LINKED ISSUE #1846: ParentBasedSampler should not make decision based on Links
The current ParentBasedSampler considers parent *and* linked activities for making sampling decision.
https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/Trace/ParentBasedSampler.cs#L56-L69

This is not conforming to the spec for ParentBasedSampler, and must be treated as a bug. The spec does allow Samplers to access Links and make decisions based on that, but `ParentBasedSampler` is not expected to do that.

Related PR: https://github.com/open-telemetry/opentelemetry-dotnet/pull/1727

### PR DESCRIPTION
**Pull request by mbakalov**
Fixes #1846.

## Changes

Previously we had functionality in `ParentBasedSampler` that would consider activity links and return RecordAndSample if one of the links was sampled. This isn't according to the [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/sdk.md#parentbased) and is removed in this PR.

In a [related PR](https://github.com/open-telemetry/opentelemetry-dotnet/pull/1727) we've added customization capability to `ParentBasedSampler` that can be used to examine links if needed.

* [ ] `CHANGELOG.md` updated for non-trivial changes

### CODE REVIEW

**Comment by mbakalov** [comment]
Just waiting for the [related PR](https://github.com/open-telemetry/opentelemetry-dotnet/pull/1727) to get merge before making changes to the actual sampler.

File: src/OpenTelemetry/CHANGELOG.md [hunk 1]
```diff
@@ -20,6 +20,10 @@ please check the latest changes
 * Added new constructor with optional parameters to allow customization of
   `ParentBasedSampler` behavior. ([#1727](https://github.com/open-telemetry/opentelemetry-dotnet/pull/1727))
 
+* Fixed [#1846](https://github.com/open-telemetry/opentelemetry-dotnet/issues/1846):
+  `ParentBasedSampler`   will no longer explicitly consider Activity links.
```
Comment (reyang) [inline src/OpenTelemetry/CHANGELOG.md:24]:
```suggestion
  `ParentBasedSampler` will no longer explicitly consider Activity links.
```

**Review by reyang — APPROVED** [review]
LGTM.

File: src/OpenTelemetry/CHANGELOG.md (hunk 1 above)
Comment (mbakalov) [reply src/OpenTelemetry/CHANGELOG.md]: Thanks - fixed!

**Review by cijothomas — APPROVED** [review]

**Comment by cijothomas** [comment]
@mbakalov thanks! please resolve conflict and we are good to merge.

**Comment by mbakalov** [comment]
> 
> 
> @mbakalov thanks! please resolve conflict and we are good to merge.

Thanks @cijothomas  - done now!

**Pull request merged**